api/api.py

```diff
@@ -1,7 +1,53 @@
-from api import app
-from flask import jsonify
+import hmac
+import hashlib
+
+from api import app, db, models
+from flask import jsonify, request
+
+
+def verify_hmac_hash(data, signature):
+    mac = hmac.new(app.config.get("GITHUB_SECRET").encode('utf-8'), data.encode('utf-8'), hashlib.sha256)
+    return hmac.compare_digest('sha256=' + mac.hexdigest(), str(signature))
 
 
 @app.route("/status")
 def status():
     return jsonify(status="ok")
+
+
+@app.route("/github-webhook", methods=["POST"])
+def github_webhook():
+    signature = request.headers.get("X-Hub-Signature-256")
+    data = request.get_data(as_text=True)
+
+    if verify_hmac_hash(data, signature):
+        if request.headers.get("X-Github-Event") == "ping":
```

Comment:
Should we maybe wrap this in a try/except to ensure we at least always return a non-500?